src/core/Form/Label/Label.tsx

```diff
@@ -40,42 +60,70 @@ const labelTextClassNames = {
 const StyledLabel = styled(
   ({
     className,
+    style,
+    contentClassName,
     theme,
     labelMode = 'visible',
-    labelSpanProps = { className: undefined },
+    wrapperProps,
     children,
     asProp = 'label',
     optionalText,
+    tooltipComponent: tooltipComponentProp,
     ...passProps
-  }: LabelProps & SuomifiThemeProp) => (
-    <HtmlDiv
-      {...(asProp ? { as: asProp } : {})}
-      className={classnames(className, baseClassName)}
-      {...passProps}
-    >
-      {labelMode === 'hidden' ? (
-        <VisuallyHidden>
-          {children}
-          {optionalText && `(${optionalText})`}
-        </VisuallyHidden>
-      ) : (
-        <HtmlSpan
-          {...labelSpanProps}
-          className={classnames(
-            labelTextClassNames.labelSpan,
-            labelSpanProps.className,
-          )}
-        >
-          {children}
-          {optionalText && (
-            <HtmlSpan className={labelTextClassNames.optionalText}>
-              {` (${optionalText})`}
+  }: LabelProps & SuomifiThemeProp) => {
+    const [wrapperRef, setWrapperRef] = useState<HTMLDivElement | null>(null);
+
+    function getTooltipComponent(
+      tooltipComponent: ReactElement | undefined,
+    ): ReactNode {
+      if (isValidElement(tooltipComponent)) {
+        return cloneElement(tooltipComponent, {
+          anchorElement: wrapperRef,
+          // trick to force tooltip to rerender every time when label changes.
+          key: Date.now(),
```

Comment:
This needs further investigation. In cases the Label is displayed inline, and it's children change, re-render might not be automatically triggered and the indication arrow position is not updated. Optimal solution would be to leave out the key, but if it does not work in all cases, maybe use something like derived state and set an uuid everytime children change?

Reply:
We will leave it like this for now.